### DISCUSSION TITLE
How do I get the price for any post code?

### DISCUSSION THREAD

**Discussion by tagpro** [amberelectric/public-api#104] [Q&A] [answered]
Hi folks, noob Amber API consumer. 

I am trying to create a system where I can use any postcode to fetch the current price. 

Can I get a list of sites from an API or something which can help me find out what the price is for a current postcode? Or if there is an existing API/library that can help me fetch that?

Thanks

**Comment by madpilot** [comment, marked as answer]
Hey Jaspreet,

No we don’t. Post code doesn’t map to price very well, as there are networks that co-exist on the same post code.

The API as it stands only returns the price for your property.

What are you planning on doing?

----------------------------------------------
Myles Eftos
Mobile: +61-409-293-183

**Reply by tagpro** [reply]
Hi, thanks. What about the API for price for an Address? I believe that can be a public API 🤔.

I am trying to build a system that can allow someone to start/stop their EV car charging based on the current price of Amber. Asking to get a token to fetch the current price of their house is not exactly ideal in the sense that the amber API returns a list of houses and if multiple locations are present for a user but the car is only in the one house, it is hard to find out which house has the car in it.

**Reply by tagpro** [reply]
It is a pet project, but I can see others using it with EV market rising at a good pace. It will also allow to reduce load with EV charging on the grid at peak hours where demand is high.

**Reply by madpilot** [reply]
> Hi, thanks. What about the API for price for an Address? I believe that can be a public API

Actually, we aren't allowed to do this (Doing address to NMI searches is a whole thing and we are limited by what we can expose to non-market participants)

We will be creating a partner API that will do something _similar_ (by NMI, or via tariff code), that may cover your use case.

> Asking to get a token to fetch the current price of their house is not exactly ideal in the sense that the amber API returns a list of houses and if multiple locations are present for a user but the car is only in the one house, it is hard to find out which house has the car in it.

We are going to be adding site nicknames soon, that might actually solve your problem? ie you can expose the site nickname, and users can work out which house there are talking about.

**Reply by tagpro** [reply]
Cool, thanks. That might solve it.
